In [17]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

df_cardio = pd.read_csv('./data/cardio_train.csv', sep = ';')
df_cardio

id    age  gender  height  weight  ap_hi  ap_lo  cholesterol  gluc  \
0          0  18393       2     168    62.0    110     80            1     1   
1          1  20228       1     156    85.0    140     90            3     1   
2          2  18857       1     165    64.0    130     70            3     1   
3          3  17623       2     169    82.0    150    100            1     1   
4          4  17474       1     156    56.0    100     60            1     1   
...      ...    ...     ...     ...     ...    ...    ...          ...   ...   
69995  99993  19240       2     168    76.0    120     80            1     1   
69996  99995  22601       1     158   126.0    140     90            2     2   
69997  99996  19066       2     183   105.0    180     90            3     1   
69998  99998  22431       1     163    72.0    135     80            1     2   
69999  99999  20540       1     170    72.0    120     80            2     1   

       smoke  alco  active  cardio  
0          0     0       1       0  
1          0     0       1       1  
2          0     0       0       1  
3          0     0       1       1  
4          0     0       0       0  
...      ...   ...     ...     ...  
69995      1     0       1       0  
69996      0     0       1       1  
69997      0     1       0       1  
69998      0     0       0       1  
69999      0     0       1       0  

[70000 rows x 13 columns]

In [5]:
df_cardio['ap_lo'].min(), df_cardio['ap_hi'].min(), df_cardio['ap_lo'].max(), df_cardio['ap_hi'].max()

(-70, -150, 11000, 16020)

## Metrics overview
- ap_hi is Systolic blood pressure. Normal 120 or little below, max 200, considered low under 90. Hard to find realistic lowest viable value, maybe 60 ? In some extreme case some humans can survive up to 300.
- ap_lo Diastolic blood pressure Normal 80,  Max 140, considered low under 60 unrealistic under 40 ?
- height 251 max

In [16]:
df_cardio['age_years'] = df_cardio['age'].apply(lambda x : x/365)
df_cardio['age_years'].min(), df_cardio['age_years'].max()
#Age range seems all good

(29.583561643835615, 64.96712328767123)

In [20]:
df_cardio_hi = df_cardio.query('ap_hi > 300')
df_cardio_lo = df_cardio.query('ap_hi < 60')
df_cardio_max_height = df_cardio.query('height < 80')
df_cardio_max_height
#No very small persons, associated weight indicates data error.

id    age  gender  height  weight  ap_hi  ap_lo  cholesterol  gluc  \
224      309  21800       2      76    55.0    120     80            1     1   
7598   10843  14661       2      70    72.0    120      8            1     1   
12770  18218  19594       1      75   168.0    120     80            1     1   
13265  18928  22456       2      71    68.0    120     80            3     1   
14323  20459  22005       1      67    57.0    120     90            1     1   
15167  21686  15812       1      70    68.0    120     80            1     1   
16699  23859  19680       2      74    98.0    140     90            1     1   
22542  32207  14538       1      68    65.0    100     60            1     1   
22723  32456  23386       1      55    81.0    130     90            1     1   
27603  39462  20978       1      64    61.0    130     70            1     1   
29157  41661  19088       1      60    69.0    110     70            1     1   
32098  45832  15374       1      72    74.0    150     90            1     1   
33607  48009  19709       2      65    72.0    130     80            1     1   
44490  63545  19120       1      65    60.0    120     80            1     1   
46319  66161  21006       2      68    71.0    120     80            1     1   
47352  67631  23297       1      75    75.0    120     80            1     1   
50789  72476  14499       2      67    60.0    110     80            1     1   
51459  73386  15432       2      70    69.0    120     80            1     1   
53344  76116  20541       2      67    80.0    120     80            1     1   
64115  91523  18426       1      59    57.6    125     67            1     1   
64454  92008  21348       1      66    63.0     12     80            1     1   
66643  95141  18830       1      57    61.0    130     90            1     1   

       smoke  alco  active  cardio  
224        0     0       1       0  
7598       0     0       1       0  
12770      1     0       1       1  
13265      0     0       1       0  
14323      0     0       1       1  
15167      0     0       0       0  
16699      0     0       1       1  
22542      0     0       0       0  
22723      0     0       1       1  
27603      0     0       1       0  
29157      0     0       0       0  
32098      0     0       1       1  
33607      0     0       0       0  
44490      0     0       1       0  
46319      0     0       1       0  
47352      0     0       1       0  
50789      1     1       1       0  
51459      0     0       0       0  
53344      0     0       0       1  
64115      0     0       0       0  
64454      0     0       0       1  
66643      0     0       1       1